File: frontend/lib/ndaFields.ts
import type { NdaFormData } from "@/types/nda";

// Mirrors backend/app/nda_fields.py's NDA_FIELDS keys and ordering — if
// NdaFormData changes, update both.
export const NDA_FIELD_LABELS: Record<keyof NdaFormData, string> = {
  partyAName: "Party A — Full Legal Name",
  partyAAddress: "Party A — Address",
  partyAEmail: "Party A — Email",
  partyBName: "Party B — Full Legal Name",
  partyBAddress: "Party B — Address",
  partyBEmail: "Party B — Email",
  businessPurpose: "Business Purpose",
  effectiveDate: "Effective Date",
  governingState: "Governing State",
  confidentialityTerm: "Confidentiality Term (years)",
  disputeCountyState: "Dispute Resolution Jurisdiction",
  disputeNoticeDays: "Dispute Notice Period (days)",
};

export const NDA_FIELD_ORDER = Object.keys(
  NDA_FIELD_LABELS
) as (keyof NdaFormData)[];

File: backend/app/nda_fields.py
"""Canonical Mutual NDA field keys and descriptions.

Mirrors `frontend/types/nda.ts`'s `NdaFormData` shape — if that changes, update
this too.
"""

NDA_FIELDS: dict[str, str] = {
    "partyAName": "Full legal name of Party A, the disclosing party",
    "partyAAddress": "Mailing address of Party A",
    "partyAEmail": "Email address of Party A",
    "partyBName": "Full legal name of Party B, the receiving party",
    "partyBAddress": "Mailing address of Party B",
    "partyBEmail": "Email address of Party B",
    "businessPurpose": "The business purpose for which confidential information will be exchanged",
    "effectiveDate": "The effective date of the agreement, as YYYY-MM-DD",
    "governingState": "The US state whose law governs the agreement",
    "confidentialityTerm": "Confidentiality term in years, a whole number",
    "disputeCountyState": "County and state for dispute resolution jurisdiction, e.g. 'Santa Clara County, California'",
    "disputeNoticeDays": "Dispute notice period in days, a whole number",
}
